src/app/store/system-config/system-config.actions.ts

```diff
@@ -0,0 +1,11 @@
+import { createAction, props } from '@ngrx/store';
+import { AdvancedConfig } from 'app/interfaces/advanced-config.interface';
+import { SystemGeneralConfig } from 'app/interfaces/system-config.interface';
+
+export const systemConfigLoaded = createAction(
+  '[System Config API] Loaded',
+  props<{ generalConfig: SystemGeneralConfig; advancedConfig: AdvancedConfig }>(),
+);
+
+export const generalConfigUpdated = createAction('[System Config] General Config Updated');
```

Comment:
These names could be improved I feel like. They should be actions not actions already performed. E.g., `reloadGeneralConfig` and `reloadAdvancedConfig`.

Reply:
Even though they are called actions, these are really events as mentioned here: https://ngrx.io/guide/store/actions#writing-actions

> Event-Driven - capture events not commands as you are separating the description of an event and the handling of that event.

Reply:
These action names are incorrect in a sense that they should be more specific, like `guiFormSubmitted`. Current ones are too general.
If you are interested in more about this: https://www.youtube.com/watch?v=JmnsEvoy-gY